wrangler-core/src/test/java/co/cask/wrangler/lineage/LineageTest.java

```diff
@@ -0,0 +1,473 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.wrangler.lineage;
+
+import co.cask.wrangler.TestingRig;
+import co.cask.wrangler.api.Executor;
+import co.cask.wrangler.api.RecipeParser;
+import co.cask.wrangler.api.lineage.LineageGenerationException;
+import co.cask.wrangler.api.lineage.MutationDefinition;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class LineageTest {
+  private static final boolean PRINT = false;
+
+  private void assertBranchingStepNode(FieldLevelLineage.BranchingTransformStepNode node,
+                                       int stepNumber, boolean back, boolean forward,
+                                       Map<String, Integer> impacting, Map<String,Integer> impacted) {
+    Assert.assertEquals(node.getTransformStepNumber(), stepNumber);
+    Assert.assertEquals(node.continueBackward(), back);
+    Assert.assertEquals(node.continueForward(), forward);
+    Assert.assertEquals(node.getImpactingBranches(), impacting);
+    Assert.assertEquals(node.getImpactedBranches(), impacted);
+  }
+
+  private void print(WranglerFieldLevelLineage lineage, List<String> inputSchema, List<String> outputSchema) {
```

Comment:
Add a comment saying this is a utility function to debug. and don't use it anywhere.